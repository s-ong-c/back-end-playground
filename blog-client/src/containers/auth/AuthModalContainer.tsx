import * as React from 'react';
import AuthModal from '../../components/auth/AuthModal';
import AuthForm from '../../components/auth/AuthForm';
import { connect } from 'react-redux';
import { RootState } from '../../modules';
import { closeAuthModal, AuthMode, changeAuthModalMode } from '../../modules/core';

interface OwnProps{};
interface StateProps{
    visible: boolean;
    mode: AuthMode;
};
interface DispatchProps{
    closeAuthModal: typeof closeAuthModal;
    changeAuthModalMode: typeof changeAuthModalMode;
};
type AuthModalContainerProps = OwnProps & StateProps & DispatchProps;

const { useCallback } = React;
const AuthModalContainer: React.SFC<AuthModalContainerProps> = ({
    visible, 
    mode,
    closeAuthModal, 
    changeAuthModalMode,
}) => {
    const onClose = useCallback(() => {
        closeAuthModal();
    }, []);
    const onToggleMode = useCallback(() => {
        const nextMode = mode === 'REGISTER' ? 'LOGIN' : 'REGISTER';
        changeAuthModalMode(nextMode);
    }, [mode]);
    return (
        <AuthModal visible={visible} onClose={onClose}>
            <AuthForm mode={mode} onToggleMode={onToggleMode} />
        </AuthModal>
        );
    };
export default connect<StateProps, DispatchProps, OwnProps, RootState>(
   state => ({
     visible: state.core.auth.visible,
     mode: state.core.auth.mode,
   }),
 { closeAuthModal, changeAuthModalMode },
)(AuthModalContainer);