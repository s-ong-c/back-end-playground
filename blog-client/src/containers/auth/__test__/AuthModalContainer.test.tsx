import React from 'react';
import { render, fireEvent} from 'react-testing-library';
import configureMockStore from 'redux-mock-store';
import AuthModalContainer from '../AuthModalContainer';
import rootReducer, { RootState } from '../../../modules';
import { Provider } from 'react-redux';

describe('AuthForm',() => {
    const setupStore = () => {
        const state = rootReducer(undefined, { type: '@@INIT'});
        const mockstore = configureMockStore<RootState>([]);
        return mockstore(state);
    }
    it('renders correctly', () => {
        const store = setupStore();
        const { container} = render(
            <Provider store={store}>
                <AuthModalContainer />
            </Provider>,
        );
        expect(container).toMatchSnapshot();
    });
})