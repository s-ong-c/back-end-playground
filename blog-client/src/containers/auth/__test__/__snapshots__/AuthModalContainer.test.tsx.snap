// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthForm renders correctly 1`] = `<div />`;
